src/app/server/Server.h

```diff
@@ -59,11 +62,119 @@ using ServerTransportMgr = chip::TransportMgr<chip::Transport::UDP
 #endif
                                               >;
 
+struct ServerInitParams
+{
+    ServerInitParams()          = default;
+    virtual ~ServerInitParams() = default;
+
+    // Not copyable
+    ServerInitParams(const ServerInitParams &) = delete;
+    ServerInitParams & operator=(const ServerInitParams &) = delete;
+
+    // Application delegate to handle some commissioning lifecycle events
+    AppDelegate * appDelegate = nullptr;
+    // Port to use for Matter commissioning/operational traffic
+    uint16_t operationalServicePort = CHIP_PORT;
+    // Port to use for UDC if supported
+    uint16_t userDirectedCommissioningPort = CHIP_UDC_PORT;
+    // Interface on which to run daemon
+    Inet::InterfaceId interfaceId = Inet::InterfaceId::Null();
+
+    // Persistent storage delegate: MUST be injected. Used to maintain storage by much common code.
+    // Must be initialized before being provided.
+    PersistentStorageDelegate * persistentStorageDelegate = nullptr;
+    // Group data provider: MUST be injected. Used to maintain critical keys such as the Identity
+    // Protection Key (IPK) for CASE. Must be initialized before being provided.
+    Credentials::GroupDataProvider * groupDataProvider = nullptr;
+    // Access control delegate: MUST be injected. Used to look up access control rules. Must be
+    // initialized before being provided
+    Access::AccessControl::Delegate * accessDelegate = nullptr;
+};
+
+/**
+ * Transitional version of ServerInitParams to assist SDK integrators in
+ * transitioning to injecting product/platform-owned resources. This version
+ * of `ServerInitParams` statically owns and initializes (via the
+ * `InitializeStaticResourcesBeforeServerInit()` method). This is to reduce the
```

Comment:
Initializes what?  Maybe:
```suggestion
 * `InitializeStaticResourcesBeforeServerInit()` method) the persistent storage delegate, the group data provider, and the access control delegate. This is to reduce the
```
and then fix up the whitespace/line-length/etc?